examples/src/test/kotlin/AggregatesBuilderTest.kt

```diff
@@ -0,0 +1,939 @@
+
+        val orders = listOf(Order(1, 1, "item1", 5), Order(2, 1, "item2", 5), Order(3, 1, "item3", 5))
+        val inventory= listOf(Inventory(1, "item1", 10), Inventory(2, "item2", 20), Inventory(3, "item3", 30))
+
+        orderCollection.insertMany(orders)
+        warehouseCollection.insertMany(inventory)
+        // TODO: Confirm why using Filters doesn't populate the stockData field? aren't these equivalent?
+        // Filters.and(
+        //  Filters.eq("order_item", Inventory::stockItem.name),
+        //  Filters.gte(Inventory::inStock.name, "\$\$order_qty")
+        //)
+
+        // :snippet-start: lookup-full-join
+        val variables = listOf(
+            Variable("order_item", "\$item"),
+            Variable("order_qty", "\$ordered")
+        )
+        val pipeline = listOf(
+            Aggregates.match(
+                Filters.expr(
+                    Document("\$and", listOf(
+                        Document("\$eq", listOf("$\$order_item", "\$${Inventory::stockItem.name}")),
+                        Document("\$gte", listOf("\$${Inventory::inStock.name}", "$\$order_qty"))
+                    ))
+                )
+            ),
+            Aggregates.project(
+                Projections.fields(
+                    Projections.exclude(Order::customerId.name, Inventory::stockItem.name),
+                    Projections.excludeId()
+                )
+            )
+        )
+        val innerJoinLookup =
+            Aggregates.lookup("warehouses", variables, pipeline, Results::stockData.name)
+        // :snippet-end:
+        val resultsCollection = database.getCollection<Results>("orders")
+        val result = resultsCollection.aggregate<Results>(listOf(innerJoinLookup)).toList()
+        val expected = listOf(
+            Results("item1", 5, listOf(StockData(10))),
+            Results("item2", 5, listOf(StockData(20))),
+            Results("item3", 5, listOf(StockData(30)))
+        )
+        assertEquals(expected, result)
+
+        // cleanup
+        warehouseCollection.drop()
+        orderCollection.drop()
+    }
+
+    @Test
+    fun groupTest() = runBlocking {
+        data class Order(@BsonId val id: Int, val customerId: Int, val item: String, val ordered: Int)
+
+        val orderCollection = database.getCollection<Order>("orders")
+        val orders = listOf(Order(1, 1, "Item1", 1), Order(2, 1, "Item2", 10), Order(3, 2, "Item3", 5), Order(4, 2, "Item4" , 50))
+        orderCollection.insertMany(orders)
+
+        data class Results(val totalQuantity: Int, val averageQuantity: Double)
+
+        val grouping = orderCollection.aggregate<Results>(listOf(
+            // :snippet-start: group
+            Aggregates.group("\$${Order::customerId.name}",
+                sum(Results::totalQuantity.name, "\$${Order::ordered.name}"),
+                avg(Results::averageQuantity.name, "\$${Order::ordered.name}")
+            )
```

Comment:
again includes results data class w/o defining it. pls replace w string literals.

same comment applies to any other instances of this pattern

Reply:
Updated throughout